pkg/errors/gatherer.go

```diff
@@ -44,6 +47,32 @@ func (g *Gatherer) Failed() <-chan struct{} {
 	return g.failed
 }
 
+// DoneOrFailed closes the returned channel when either:
+// all routines returned successfully OR
+// any routine returned with an error.
+func (g *Gatherer) DoneOrFailed() <-chan struct{} {
+	return g.DoneOrFailedCtx(context.Background())
+}
+
+// DoneOrFailedCtx closes the returned channel when either:
+// all routines returned successfully OR
+// any routine returned with an error OR
+// the context was cancelled.
+func (g *Gatherer) DoneOrFailedCtx(ctx context.Context) <-chan struct{} {
+	ret := make(chan struct{})
+
+	go func() {
+		select {
+		case <-g.failed:
+		case <-g.wg.WaitCh():
+		case <-ctx.Done():
+		}
+		close(ret)
+	}()
+
+	return ret
+}
```

Comment:
*(Cannot leave comment on unchanged lines)*
Also add a `WaitCtx(Context) bool` function that calls the waitgroup's `WaitCtx()`.

Reply:
Might also return `error, bool`, but that's a bit clumsy IMO. Either way is fine with me.

Reply:
Would like to not blow up the MR, referenced in https://github.com/perun-network/poly-go/issues/5